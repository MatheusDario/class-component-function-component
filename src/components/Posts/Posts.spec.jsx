import { screen } from '@testing-library/react';
import { Posts } from '.';
import { renderTheme } from '../../styles/render-theme';

const props = {
  posts: [
    {
      id: 1,
      title: 'Title 1',
      body: 'Body 1',
      cover: 'images/fundo.png',
    },
    {
      id: 2,
      title: 'Title 2',
      body: 'Body 2',
      cover: 'images/fundo.png',
    },
    {
      id: 3,
      title: 'Title 3',
      body: 'Body 3',
      cover: 'images/fundo.png',
    },
  ],
};

describe('<Posts />', () => {
  it('should render posts component', () => {
    renderTheme(<Posts {...props} />);
    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
    expect(screen.getByRole('img', { name: 'Title 3' })).toHaveAttribute(
      'src',
      'images/fundo.png',
    );
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Posts {...props} />);
    expect(container).toMatchSnapshot();
  });
});
