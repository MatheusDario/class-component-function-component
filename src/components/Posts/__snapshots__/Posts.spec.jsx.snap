// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 30px;
}

.c1 {
  background-color: #cecece;
  box-shadow: 0px 0px 10px rgba(255, 255, 255, .4 );
  border-radius: 10px;
  margin: 1rem 0;
  transition: all 300ms ease-in-out;
  max-width: 50vh;
}

.c1:hover {
  transform: scale(1.05);
}

.c2 {
  padding: 1.5rem;
}

.c4 {
  font-size: 2.3rem;
}

.c5 {
  text-align: justify;
  font-size: 1.3rem;
}

.c3 {
  width: 100%;
}

<div>
  <main
    class="c0"
  >
    <section
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Title 1"
          class="c3"
          src="images/fundo.png"
        />
        <h1
          class="c4"
        >
          Title 1
        </h1>
        <p
          class="c5"
        >
          Body 1
        </p>
      </div>
    </section>
    <section
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Title 2"
          class="c3"
          src="images/fundo.png"
        />
        <h1
          class="c4"
        >
          Title 2
        </h1>
        <p
          class="c5"
        >
          Body 2
        </p>
      </div>
    </section>
    <section
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Title 3"
          class="c3"
          src="images/fundo.png"
        />
        <h1
          class="c4"
        >
          Title 3
        </h1>
        <p
          class="c5"
        >
          Body 3
        </p>
      </div>
    </section>
  </main>
</div>
`;
